kong/plugins/standard-webhooks/internal.lua

```diff
@@ -0,0 +1,70 @@
+local kong = kong
+local hmac = require "resty.openssl.hmac"
+local tonumber = tonumber
+local ngx = ngx
+
+local HEADER_WEBHOOK_ID = "webhook-id"
+local HEADER_WEBHOOK_SIGN = "webhook-signature"
+local HEADER_WEBHOOK_TS = "webhook-timestamp"
+
+local function sign(secret, id, ts, payload)
+  local d, err = hmac.new(secret, "sha256")
+  if err then
+    kong.log.error(err)
+    return kong.response.error(500)
+  end
+  d:update(id .. "." .. ts .. "." .. payload)
+  local r, err = d:final()
+  if err then
+    kong.log.error(err)
+    return kong.response.error(500)
+  end
+  return "v1," .. ngx.encode_base64(r)
+end
+
+local function extract_webhook()
+  local headers = kong.request.get_headers()
+
+  local id = headers[HEADER_WEBHOOK_ID]
+  local signature = headers[HEADER_WEBHOOK_SIGN]
+  local ts = headers[HEADER_WEBHOOK_TS]
```

Comment:
we need to catch the case when those headers appear more than one time (and they will be a table instead of string)
```lua
      if type(id) == "table" then
        id = id[1]
      end
      -- ... others
```

Reply:
updated